samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -84,7 +85,7 @@ public void kill() {
     // since currently we only support single actual remote job, we can get its status without
     // building the execution plan.
     try {
-      JobConfig jc = new JobConfig(appDesc.getConfig());
+      JobConfig jc = new JobConfig(JobConfigUtil.generateJobIdAndName(appDesc.getConfig()));
```

Comment:
This JobConfigUtil.generateJobIdAndName should be in the planner. You can add a function directly there like 
   planer.createSingleJobConfig(). This will be better handled in a single place.

Reply:
will do